### PR TITLE
backup: fix issue #2028

### LINKED ISSUE #2028: Retrieve cluster and client TLS config from TidbCluster
## Feature Request

**Is your feature request related to a problem? Please describe:**

Retrieve cluster and client TLS config from TidbCluster
**Describe the feature you'd like:**

Backup controller retrieves the cluster and client TLS config from TidbCluster and then passes them to the backup-manager job as command-line options so that users do not have to configure them explicitly in the Backup and Restore CR.
**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by shuijing198799**
### What problem does this PR solve? 
fix #2028
### What is changed and how does it work?
add tclister in backup and restore controller and delete cluster tls in BRCONFIG
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change

### CODE REVIEW

File: cmd/backup-manager/app/cmd/backup.go [hunk 1]
```diff
@@ -42,6 +42,8 @@ func NewBackupCommand() *cobra.Command {
 
 	cmd.Flags().StringVar(&bo.Namespace, "namespace", "", "Backup CR's namespace")
 	cmd.Flags().StringVar(&bo.ResourceName, "backupName", "", "Backup CRD object name")
+	cmd.Flags().BoolVar(&bo.TLSClient, "client-tls", false, "trigger of client tls")
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/cmd/backup.go]:
```suggestion
	cmd.Flags().BoolVar(&bo.TLSClient, "client-tls", false, "Whether client tls is enabled")
```

File: cmd/backup-manager/app/cmd/backup.go [hunk 2]
```diff
@@ -42,6 +42,8 @@ func NewBackupCommand() *cobra.Command {
...
+	cmd.Flags().BoolVar(&bo.TLSCluster, "cluster-tls", false, "trigger of cluster tls")
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/cmd/backup.go]:
```suggestion
	cmd.Flags().BoolVar(&bo.TLSCluster, "cluster-tls", false, "Whether cluster tls is enabled")
```

File: cmd/backup-manager/app/cmd/restore.go [hunk 3]
```diff
@@ -44,6 +44,8 @@ func NewRestoreCommand() *cobra.Command {
 
 	cmd.Flags().StringVar(&ro.Namespace, "namespace", "", "Restore CR's namespace")
 	cmd.Flags().StringVar(&ro.ResourceName, "restoreName", "", "Restore CRD object name")
+	cmd.Flags().BoolVar(&ro.TLSClient, "tls-client", false, "trigger of client tls")
+	cmd.Flags().BoolVar(&ro.TLSCluster, "tls-cluster", false, "trigger of cluster tls")
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/cmd/restore.go]: Please keep consistent with the parameters of `backup`.

File: pkg/backup/backup/backup_manager.go [hunk 4]
```diff
@@ -255,6 +259,14 @@ func (bm *backupManager) makeExportJob(backup *v1alpha1.Backup) (*batchv1.Job, s
 func (bm *backupManager) makeBackupJob(backup *v1alpha1.Backup) (*batchv1.Job, string, error) {
 	ns := backup.GetNamespace()
 	name := backup.GetName()
+	backupNamespace := ns
+	if backup.Spec.BR.ClusterNamespace != "" {
+		backupNamespace = backup.Spec.BR.ClusterNamespace
+	}
+	tc, err := bm.tcLister.TidbClusters(backupNamespace).Get(backup.Spec.BR.Cluster)
+	if err != nil {
+		return nil, fmt.Sprintf("failed to fetch tidbcluster %s/%s", backup.Spec.BR.Cluster, backupNamespace), err
```
Comment (DanielZhangQD) [inline pkg/backup/backup/backup_manager.go]:
```suggestion
		return nil, fmt.Sprintf("failed to fetch tidbcluster %s/%s", backupNamespace, backup.Spec.BR.Cluster), err
```

File: pkg/backup/restore/restore_manager.go [hunk 5]
```diff
@@ -240,6 +244,14 @@ func (rm *restoreManager) makeImportJob(restore *v1alpha1.Restore) (*batchv1.Job
 func (rm *restoreManager) makeRestoreJob(restore *v1alpha1.Restore) (*batchv1.Job, string, error) {
 	ns := restore.GetNamespace()
 	name := restore.GetName()
+	restoreNamespace := ns
+	if restore.Spec.BR.ClusterNamespace != "" {
+		restoreNamespace = restore.Spec.BR.ClusterNamespace
+	}
+	tc, err := rm.tcLister.TidbClusters(restoreNamespace).Get(restore.Spec.BR.Cluster)
+	if err != nil {
+		return nil, fmt.Sprintf("failed to fetch tidbcluster %s/%s", restore.Spec.BR.Cluster, restoreNamespace), err
```
Comment (DanielZhangQD) [inline pkg/backup/restore/restore_manager.go]:
```suggestion
		return nil, fmt.Sprintf("failed to fetch tidbcluster %s/%s", restoreNamespace, restore.Spec.BR.Cluster), err
```

File: cmd/backup-manager/app/cmd/backup.go (hunk 1 above)
Comment (shuijing198799) [reply cmd/backup-manager/app/cmd/backup.go]: done

File: cmd/backup-manager/app/cmd/backup.go (hunk 2 above)
Comment (shuijing198799) [reply cmd/backup-manager/app/cmd/backup.go]: done

File: cmd/backup-manager/app/cmd/restore.go (hunk 3 above)
Comment (shuijing198799) [reply cmd/backup-manager/app/cmd/restore.go]: done

File: pkg/backup/backup/backup_manager.go (hunk 4 above)
Comment (shuijing198799) [reply pkg/backup/backup/backup_manager.go]: done

File: pkg/backup/restore/restore_manager.go (hunk 5 above)
Comment (shuijing198799) [reply pkg/backup/restore/restore_manager.go]: done

File: cmd/backup-manager/app/cmd/restore.go [hunk 6]
```diff
@@ -44,6 +44,8 @@ func NewRestoreCommand() *cobra.Command {
 
 	cmd.Flags().StringVar(&ro.Namespace, "namespace", "", "Restore CR's namespace")
 	cmd.Flags().StringVar(&ro.ResourceName, "restoreName", "", "Restore CRD object name")
+	cmd.Flags().BoolVar(&ro.TLSClient, "tls-client", false, "Whether client tls is enabled")
+	cmd.Flags().BoolVar(&ro.TLSCluster, "tls-cluster", false, "Whether cluster tls is enabled")
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/cmd/restore.go]: The option names are incorrect.

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 7]
```diff
@@ -850,10 +850,6 @@ type BackupSpec struct {
 // +k8s:openapi-gen=true
 // BRConfig contains config for BR
 type BRConfig struct {
-	// Whether enable the TLS connection between TiDB server components
-	// Optional: Defaults to nil
-	// +optional
-	TLSCluster *TLSCluster `json:"tlsCluster,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go]: Please also update the examples.

File: cmd/backup-manager/app/cmd/backup.go [hunk 8]
```diff
@@ -42,6 +42,8 @@ func NewBackupCommand() *cobra.Command {
 
 	cmd.Flags().StringVar(&bo.Namespace, "namespace", "", "Backup CR's namespace")
 	cmd.Flags().StringVar(&bo.ResourceName, "backupName", "", "Backup CRD object name")
+	cmd.Flags().BoolVar(&bo.TLSClient, "client-tls", false, "Whether of client tls is enabled")
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/cmd/backup.go]:
```suggestion
	cmd.Flags().BoolVar(&bo.TLSClient, "client-tls", false, "Whether client tls is enabled")
```

File: cmd/backup-manager/app/cmd/backup.go [hunk 9]
```diff
@@ -42,6 +42,8 @@ func NewBackupCommand() *cobra.Command {
...
+	cmd.Flags().BoolVar(&bo.TLSCluster, "cluster-tls", false, "Whether of cluster tls is enabled")
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/cmd/backup.go]:
```suggestion
	cmd.Flags().BoolVar(&bo.TLSCluster, "cluster-tls", false, "Whether cluster tls is enabled")
```

**Pull request merged**